test/point_test.cpp

```diff
@@ -315,30 +313,99 @@ TEST_CASE("Point tests", "[Point]")
         REQUIRE_THROWS(point6.ClassificationBitFields());
         REQUIRE_NOTHROW(point6.ToString());
 
-        auto point7 = Point(7);
+        auto point7 = Point();
+        point7.ToPointFormat(7);
 
-        point7.SetRgb(UINT16_MAX, UINT16_MAX, UINT16_MAX);
+        point7.Rgb(UINT16_MAX, UINT16_MAX, UINT16_MAX);
         REQUIRE(point7.R() == UINT16_MAX);
         REQUIRE(point7.G() == UINT16_MAX);
         REQUIRE(point7.B() == UINT16_MAX);
 
-        REQUIRE_THROWS(point7.SetNirVal(UINT16_MAX));
-        REQUIRE_THROWS(point7.GetNirVal());
+        REQUIRE_THROWS(point7.Nir(UINT16_MAX));
+        REQUIRE_THROWS(point7.Nir());
 
-        auto point8 = Point(8);
+        auto point8 = Point();
+        point8.ToPointFormat(8);
 
-        point8.SetNirVal(UINT16_MAX);
-        REQUIRE(point8.GetNirVal() == UINT16_MAX);
+        point8.Nir(UINT16_MAX);
+        REQUIRE(point8.Nir() == UINT16_MAX);
 
-        auto point9 = Point(9);
+        auto point9 = Point();
+        point9.ToPointFormat(9);
 
-        REQUIRE_THROWS(point9.SetRgb(UINT16_MAX, UINT16_MAX, UINT16_MAX));
+        REQUIRE_THROWS(point9.Rgb(UINT16_MAX, UINT16_MAX, UINT16_MAX));
         REQUIRE_THROWS(point9.R());
         REQUIRE_THROWS(point9.G());
         REQUIRE_THROWS(point9.B());
-        REQUIRE_THROWS(point9.SetNirVal(UINT16_MAX));
-        REQUIRE_THROWS(point9.GetNirVal());
+        REQUIRE_THROWS(point9.Nir(UINT16_MAX));
+        REQUIRE_THROWS(point9.Nir());
 
         // No need to test point10, it has all attributes and they have all been tested already
     }
+
+    SECTION("Point conversion")
+    {
+        auto point = Point();
+        point.ToPointFormat(0);
+
+        point.ReturnNumber(5);
+        point.NumberOfReturns(6);
+        point.ScanDirectionFlag(false);
+        point.EdgeOfFlightLineFlag(true);
+        point.Classification(15);
+        point.Synthetic(true);
+        point.KeyPoint(false);
+        point.Withheld(true);
+        point.ScanAngleRank(45);
+
+        point.ToPointFormat(6);
+
+        REQUIRE(point.ReturnNumber() == 5);
+        REQUIRE(point.NumberOfReturns() == 6);
+        REQUIRE(point.ScanDirectionFlag() == false);
+        REQUIRE(point.EdgeOfFlightLineFlag() == true);
+        REQUIRE(point.Classification() == 15);
+        REQUIRE(point.Synthetic() == true);
+        REQUIRE(point.KeyPoint() == false);
+        REQUIRE(point.Withheld() == true);
+        REQUIRE(point.Overlap() == false);
+        REQUIRE(point.ExtendedScanAngle() == 7500);
+        REQUIRE(point.ScannerChannel() == 0);
+
+        point.ToPointFormat(0);
+
+        REQUIRE(point.ReturnNumber() == 5);
+        REQUIRE(point.NumberOfReturns() == 6);
+        REQUIRE(point.ScanDirectionFlag() == false);
```

Comment:
can we add some REQUIRE_THROWS tests for the converted points, such as point.overlap()?